### PR TITLE
adds some new nullrods with some back end null rod refactors

### PR DESCRIPTION
**Pull request by Sishen1542**
## About The Pull Request
Adds in three (two, but fixes one) new null rods for your chaplaining pleasure! 

- Laser AK with 0.1 burn on a hitscan laser, with typical laser tag cell/cost (10 shots per full cell, recharges over time) that does 150 burn if you execution style a cultist!
Has reduced melee force in exchange for the (laughable) laser.

- Divine Lightblade, a hyper-eutactic blade fit for any anime protagonist.
20 force, 50 block, takes up both hands.

- Hypertool, a functional multitool that deals armor penetrating brain damage.
- Bo Staff now deals stamina damage on hit if you're in disarm intent. Cannot attack people if they're unconscious, in stam crit, or health crit. Also has a 25% chance to jedi spin.
- Blessed Baseball Bat. 14 force, 14 throwforce. Functional bat. BONK!

I also normalized all stamina values for the holy weapons, so they'll all take between 4-5 stamina to swing. This also comes with a regretful nerf to the holy whip, as 2 tile range proves very powerful with such a high force output. 

Also also added the destruction of runes to the bible object, since adding snowflake checks to all the holy weapons would be silly.

Also also also reworked the cultist removal bit from prayer beads into just an injection of holy water, shortened action timer in exchange for it.

## Why It's Good For The Game
Funny fun fun fun, and flavorful too. Nothing is outright gamebreaking (150 damage seems like a lot, but it's tied to the execution style timer)

Prayer beads being a 10 second timer for an instant deconvert was dumb, so I just made it give some holy water to make it take longer (and more clicks) to work.

## Changelog
:cl:
add: holy lasrifle, hypertool, divine lightblade, and blessed baseball bat. Four new holy weapons!
balance: stamina and force tweaks for most holy weapons.
tweak: prayer beads no longer deconvert after a 10 second timer, now just inject holy water for delayed effect
tweak: cultist runes are now destroyed with a bible or a null rod
tweak: bo-staff now functions as a nerfed sleepy carp staff
fix: fixed kevinz code to add in non-null rod child items as holy weapons
/:cl:

### CODE REVIEW

**Comment by Drowzyleaf** [comment]
they made https://imgur.com/t5lTLYk into a real thing

File: code/modules/antagonists/cult/runes.dm [hunk 1]
```diff
@@ -61,8 +61,8 @@ Runes can either be invoked by one's self or with many different cultists. Each
 		if(do_after(user, 15, target = src))
 			to_chat(user, "<span class='notice'>You carefully erase the [lowertext(cultist_name)] rune.</span>")
 			qdel(src)
-	else if(istype(I, /obj/item/nullrod))
-		user.say("BEGONE FOUL MAGIKS!!", forced = "nullrod")
+	else if(istype(I, /obj/item/storage/book/bible))
+		user.say("BEGONE FOUL MAGICKS!!", forced = "bible")
```
Comment (Trilbyspaceclone) [inline code/modules/antagonists/cult/runes.dm:65]: Why not make both work?

File: modular_citadel/code/game/objects/items/holy_weapons.dm [hunk 2]
```diff
@@ -30,24 +30,16 @@
 		"<span class='info'>You kneel[M == user ? null : " next to [M]"] and begin a prayer to [deity_name].</span>")
 
 	praying = TRUE
-	if(do_after(user, 100, target = M))
+	if(do_after(user, 20, target = M))
 		if(istype(M, /mob/living/carbon/human)) // This probably should not work on catpeople. They're unholy abominations.
 			var/mob/living/carbon/human/target = M
-
-			if(iscultist(M) || is_servant_of_ratvar(M)) //ripped from holywater code.
```
Comment (Trilbyspaceclone) [inline modular_citadel/code/game/objects/items/holy_weapons.dm:37]: ??? 
Out side the scope of the PR, and not needed change this worked fine

**Review by Trilbyspaceclone — CHANGES_REQUESTED** [review]

File: code/game/objects/items/holy_weapons.dm [hunk 3]
```diff
@@ -585,6 +585,7 @@
 	lefthand_file = 'icons/mob/inhands/weapons/melee_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/weapons/melee_righthand.dmi'
 	slot_flags = ITEM_SLOT_BELT
+	force = 12
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/holy_weapons.dm]: Why nerf chainhand

File: code/modules/antagonists/cult/runes.dm (hunk 1 above)
Comment (Sishen1542) [reply code/modules/antagonists/cult/runes.dm:65]: im fine with both, up to maintainers

File: modular_citadel/code/game/objects/items/holy_weapons.dm (hunk 2 above)
Comment (Sishen1542) [reply modular_citadel/code/game/objects/items/holy_weapons.dm:37]: 10 sec for deconvert vs 30+ through holy water is a little silly

**Comment by Trilbyspaceclone** [comment]
Im strongly against the rosary beads nerf, it makes its job worst.
Its ment to help people and be a non-harmful way. Adding in holywater to the proc rather then a deconvert is dumb.
Theirs no reason to change it from its 10 second deconvert, as thats the same time it takes to convert people. Its rather well balanced and rarely abused. + outside of 2 game modes is only able to do small heals that anyone with a soda despender can.
Changing it seems more of a salt nerf then anything.

**Comment by Sishen1542** [comment]
> Changing it seems more of a salt nerf then anything.

salt nerf from someone who doesn't play antags :thinking:

**Comment by Drowzyleaf** [comment]
I'd say the prayer bead change isn't horrible. Right now, they may as well be useless on non-cult, since the healing is negligible for ten seconds of prayer. Now, they can still deconvert if a bit slower with the holy water, and they can work decently well on non-cult rounds for more pacifistic chaplains to heal others.

**Comment by Sishen1542** [comment]
> I'd say the prayer bead change isn't horrible. Right now, they may as well be useless on non-cult, since the healing is negligible for ten seconds of prayer. Now, they can still deconvert if a bit slower with the holy water, and they can work decently well on non-cult rounds for more pacifistic chaplains to heal others.

it also heals damage types that the bible doesnt cover, so yeah its pretty good at that
5 oxy/toxic per tick (chem ticks are effectively 2 secs, length of the action timer) is strong

**Comment by Trilbyspaceclone** [comment]
They now CANT deconvert with beads as 5u isnt going to last 10 seconds, thus making it useless for that and only deal harm to the person they are healing

**Comment by Drowzyleaf** [comment]
well, no. The timer was reduced to 2 seconds. Won't metabolize in that time, I'm sure.

**Comment by Sishen1542** [comment]
> well, no. The timer was reduced to 2 seconds. Won't metabolize in that time, I'm sure.

the idea is you slap them a few times with it, yeah

**Comment by Trilbyspaceclone** [comment]
Unlisted change didnt, know it would be lowered to 2 seconds, im now ok with this all

**Review by nik707 — APPROVED** [review]
The code looks awesome. That said, because I'm not super up to par yet on mains balancing standards, ask Kev or Bhijn to give it a quick glance over to make sure they don't have any grievances.

**Review by silicons — APPROVED** [review]

**Pull request merged**